Cabal/Distribution/PackageDescription.hs

```diff
@@ -1193,11 +1199,30 @@ data Condition c = Var c
                  | CAnd (Condition c) (Condition c)
     deriving (Show, Eq, Typeable, Data, Generic)
 
+-- | Boolean negation of a 'Condition' value.
 cNot :: Condition a -> Condition a
 cNot (Lit b)  = Lit (not b)
 cNot (CNot c) = c
 cNot c        = CNot c
 
+-- | Boolean AND of two 'Condtion' values.
+cAnd :: Condition a -> Condition a -> Condition a
+cAnd (Lit False) _           = Lit False
+cAnd _           (Lit False) = Lit False
+cAnd (Lit True)  x           = x
+cAnd x           (Lit True)  = x
+cAnd x           y           = CAnd x y
+
+-- | Boolean OR of two 'Condition' values.
+cOr :: Eq v => Condition v -> Condition v -> Condition v
+cOr  (Lit True)  _           = Lit True
+cOr  _           (Lit True)  = Lit True
+cOr  (Lit False) x           = x
+cOr  x           (Lit False) = x
+cOr  c           (CNot d)
+  | c == d                   = Lit True
+cOr  x           y           = COr x y
```

Comment:
I forgot to mention that I added a case to my snippet:

    cor  (CNot c)    d
      | c == d                   = Lit True

Without it, the function testing whether #3199 applies doesn't ignore this use of buildable:
```
executable exe
  if !flag(build-exe)
    buildable: False
  else
    build-depends: package
  [...]
```